nixos/lib/make-ext4-fs.nix

```diff
@@ -37,32 +37,26 @@ pkgs.stdenv.mkDerivation {
       ${populateImageCommands}
       )
 
-      # Add the closures of the top-level store objects.
-      storePaths=$(cat ${sdClosureInfo}/store-paths)
+      # Create nix/store before copying path
+      mkdir -p ./rootImage/nix/store
+
+      # Also include a manifest of the closures in a format suitable for nix-store --load-db
+      cp ${sdClosureInfo}/registration ./rootImage/nix-path-registration
+
+      echo "preparing store paths for image..."
+      cp -a $(cat ${sdClosureInfo}/store-paths) -t ./rootImage/nix/store/
```

Comment:
Doesn't this risk overrunning the command line length limit? How about a `while read -r; do cp ...; done` loop or something?

Reply:
Fixed!